pkg/network/usm/utils/file_registry.go

```diff
@@ -40,170 +81,160 @@ func NewFileRegistry() *FileRegistry {
 	}
 
+	pathID := path.ID
 	r.m.Lock()
 	defer r.m.Unlock()
+	if r.stopped {
+		return
+	}
+
 	if _, found := r.blocklistByID[pathID]; found {
-		r.telemetry.libBlocked.Add(1)
+		r.telemetry.fileBlocked.Add(1)
 		return
 	}
 
 	if reg, found := r.byID[pathID]; found {
 		if _, found := r.byPID[pid][pathID]; !found {
 			reg.uniqueProcessesCount.Inc()
-			// Can happen if a new process opens the same so.
+			// can happen if a new process opens an already active file
 			if len(r.byPID[pid]) == 0 {
 				r.byPID[pid] = pathIdentifierSet{}
 			}
 			r.byPID[pid][pathID] = struct{}{}
 		}
-		r.telemetry.libAlreadyRegistered.Add(1)
+		r.telemetry.fileAlreadyRegistered.Add(1)
 		return
 	}
 
-	if err := activate(pathID, root, libPath); err != nil {
-		log.Debugf("error registering library (adding to blocklist) %s path %s by pid %d : %s", pathID.String(), hostLibPath, pid, err)
-		// we are calling UnregisterCB here as some uprobes could be already attached, UnregisterCB cleanup those entries
-		if deactivate != nil {
-			if err := deactivate(pathID); err != nil {
-				log.Debugf("UnregisterCB library %s path %s : %s", pathID.String(), hostLibPath, err)
-			}
-		}
-		// save sentinel value, so we don't attempt to re-register shared
-		// libraries that are problematic for some reason
+	if err := activationCB(path); err != nil {
+		// we are calling `deactivationCB` here as some uprobes could be already attached
+		_ = deactivationCB(FilePath{ID: pathID})
+		// add `pathID` to blocklist so we don't attempt to re-register files
+		// that are problematic for some reason
 		r.blocklistByID[pathID] = struct{}{}
-		r.telemetry.libHookFailed.Add(1)
+		r.telemetry.fileHookFailed.Add(1)
 		return
 	}
 
-	reg := r.newRegistration(deactivate)
+	reg := r.newRegistration(deactivationCB)
 	r.byID[pathID] = reg
 	if len(r.byPID[pid]) == 0 {
 		r.byPID[pid] = pathIdentifierSet{}
 	}
 	r.byPID[pid][pathID] = struct{}{}
-	log.Debugf("registering library %s path %s by pid %d", pathID.String(), hostLibPath, pid)
-	r.telemetry.libRegistered.Add(1)
+	log.Debugf("registering file %s path %s by pid %d", pathID.String(), path.HostPath, pid)
+	r.telemetry.fileRegistered.Add(1)
 }
 
-// Unregister a pid if exists, unregisterCB will be called if his uniqueProcessesCount == 0
-func (r *FileRegistry) Unregister(pid int) {
-	pidU32 := uint32(pid)
-	r.m.RLock()
-	_, found := r.byPID[pidU32]
-	r.m.RUnlock()
+// Unregister a PID if it exists
+//
+// All files that were previously referenced by the given PID will have their
+// reference counters decremented by one. For any file for the number of
+// references drops to zero, we'll execute the *deactivationCB* previously
+// supplied during the `Register` call.
+func (r *FileRegistry) Unregister(pid uint32) {
+	r.m.Lock()
+	defer r.m.Unlock()
+	if r.stopped {
+		return
+	}
+
+	_, found := r.byPID[pid]
 	if !found {
 		return
 	}
 
-	r.m.Lock()
-	defer r.m.Unlock()
-	paths, found := r.byPID[pidU32]
+	paths, found := r.byPID[pid]
```

Comment:
now that you lock on the beginning of the file, this is a redundant duplication of line 176

Reply:
Good catch